Automattic/wp-calypso
# i18n: Remove redundant argument from getLanguage()

In `getLanguage()`, the function arguments are only used like this:

https://github.com/Automattic/wp-calypso/blob/0dbf3b877aad2a17393fc3d34ec93cced21425ed/client/lib/i18n-utils/utils.js#L70-L72

In other words, the function body uses the second argument if it's specified, and falls back to the firrst if it isn't.

Arguably, this just added to the noise and made the code harder to read. This kind of operation should be up to the function's consumer, not its implementation.

This PR removes `getLanguage()`'s second argument, and changes `switchLocale()` accordingly. A potential follow-up will be to also remove the redundancy in Redux state. My thinking is that we should only store the variant (naming it `localeSlug` in the reducer should be fine tho), and modify our selectors so there's one that returns the variation, and one that returns the variation-free locale slug -- both of them however evaluating the same information in the Redux tree.

**To verify that this fix is complete:**

- Verify that the only function that made use of `getLanguage()`'s second arg was the `switchLocale()` function.
- Verify that the only consumer of `switchLocale()` is the `setLocale()` action.

**To test:**

Team Global will probably know this better, but I guess try switching the user's UI language in `/me`?
